sisl/io/sile.py

```diff
@@ -455,6 +455,13 @@ def __getattr__(self, name):
         """ Override to check the handle """
         if name == 'fh':
             raise AttributeError(f"The filehandle for {self.file} has not been opened yet...")
+
+        # For some reason siles are not picklable if one lets "_file" or "file"
+        # be searched in self.fh. In that case, trying to load siles with pickle ends up in a recursion
+        # error.
+        if name in ("_file", "file"):
+            raise AttributeError
+
```

Comment:
@zerothi do you know why this happened? Does this disturb in some way the functionality? Maybe you can solve the problem in a better way.

Reply:
? Context?

Reply:
Sorry. This snippet should reproduce the error that I described in the commented lines:

```python
import sisl
import pickle
from pathlib import Path

sile = sisl.get_sile("/path/to/sile") 
save_filename = Path.home() / "sile_attempt"

with open(save_filename, "wb") as fh:
    pickle.dump(sile, fh)
    
with open(save_filename, "rb") as fh:
    loaded = pickle.load(fh)
```

So basically, saving and loading a `Sile`, gives a recursion error when loading. This interferes with the saving and loading abalities of plots, which use `dill` currently.

But I just realized that adding these methods to `BaseSile` solves the problem (there's no need for the dirty fix that I did in this commit):

```python
def __getstate__(self):

        return self.__dict__ 

def __setstate__(self, state):
        
        self.__dict__ = state
```

Can I add them?

Also, if I try with a RHO sile, it gives a more profound error:

```
Can't pickle <class 'sisl.io.siesta.binaries.rhoSileSiesta'>: it's not found as sisl.io.siesta.binaries.rhoSileSiesta
```

Do you know how to solve this?

Reply:
I think this deserves a separate PR, could you please branch from master, and do a new PR with the changes you suggest, also amend a test, it would be nice to add a test here: `sisl/io/siesta/tests/test_grid.py` and possibly also other places. A test with the NetCDF files would also be very nice.